### PR TITLE
Allow git dependencies to be local files

### PR DESCRIPTION
**Pull request by yufeih**
#4559

### CODE REVIEW

File: src/docfx/config/Config.cs [hunk 1]
```diff
@@ -166,10 +166,9 @@ internal sealed class Config
         public readonly SourceInfo<string>[] MetadataSchema = Array.Empty<SourceInfo<string>>();
 
         /// <summary>
-        /// Get the theme repo url like https://github.com/docs/theme#master
-        /// It's used for legacy doc(docs.com) sites build
+        /// Get the template folder or git repository url (like https://github.com/docs/theme#master)
         /// </summary>
-        public readonly string Theme = string.Empty;
+        public readonly string Template = string.Empty;
```
Comment (OsmondJiang) [inline src/docfx/config/Config.cs]: that's a break change, also need change it Configuration Migration Tool, @mingwli

File: src/docfx/restore/RestoreGit.cs [hunk 2]
```diff
@@ -183,32 +183,26 @@ void AddWorkTrees()
 
         private static IEnumerable<(string remote, string branch, GitFlags flags)> GetGitDependencies(Config config, string locale, Repository rootRepository)
         {
-            var dependencies = config.Dependencies.Values.Select(url =>
+            foreach (var (_, url) in config.Dependencies)
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreGit.cs]: will this change also applies to Dependencies? How we know the branch info for local dependency folder? Can we add a spec test for that?

File: src/docfx/restore/RestoreGit.cs [hunk 3]
```diff
@@ -183,32 +183,26 @@ void AddWorkTrees()
...
-            dependencies = dependencies.Concat(GetThemeGitDependencies(config, locale));
+                if (UrlUtility.IsHttp(url))
+                {
+                    var (remote, branch, _) = UrlUtility.SplitGitUrl(url);
+                    yield return (remote, branch, GitFlags.DepthOne);
+                }
+            }
 
-            if (rootRepository != null)
+            if (UrlUtility.IsHttp(config.Template))
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreGit.cs]: only root repo need get template

File: src/docfx/restore/RestoreMap.cs [hunk 4]
```diff
@@ -177,10 +186,10 @@ public static RestoreMap
             {
                 if (!successed && root)
                 {
-                    foreach (var (k, v) in acquired)
-                    {
-                        ReleaseGit(v.git, LockType.Shared, false);
-                    }
+                foreach (var (k, v) in acquired)
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreMap.cs]: spaces

File: src/docfx/config/Config.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/config/Config.cs]: yep, intentional

File: src/docfx/restore/RestoreMap.cs [hunk 5]
```diff
@@ -21,20 +20,26 @@ public RestoreMap(IReadOnlyDictionary<(string remote, string branch, string comm
...
-        }
+            if (!UrlUtility.IsHttp(remote))
+            {
+                var fullPath = Path.Combine(docsetPath, remote);
+                if (Directory.Exists(fullPath))
+                {
+                    return (fullPath, new RestoreMap(_acquiredGits));
+                }
+
+                // TODO: Intentionally don't fallback to fallbackDocset for git restore path,
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreMap.cs]: it;s overwrite config scenario, not for test happy

**Pull request merged**